File: DTOs/GoodReciptDto/GrImportLDto.cs
using System.ComponentModel.DataAnnotations;

namespace WMS_WEBAPI.DTOs
{
    public class GrImportLDto : BaseImportLineEntityDto
    {
        public long? LineId { get; set; }
        public long HeaderId { get; set; }
    }

    public class CreateGrImportLDto : BaseImportLineCreateDto
    {
        public long? LineId { get; set; }
        
        [Required]
        public long HeaderId { get; set; }
    }

    public class UpdateGrImportLDto : BaseImportLineUpdateDto
    {
        public long? LineId { get; set; }
        public long? HeaderId { get; set; }
    }
}

File: WebApi/Models/DTOs/GoodReciptDto/GrImportLDto.ts
import { BaseImportLineCreateDto, BaseImportLineEntityDto, BaseImportLineUpdateDto } from '../../index';
export interface GrImportLDto extends BaseImportLineEntityDto {
  LineId?: number;
  HeaderId: number;
}

export interface CreateGrImportLDto extends BaseImportLineCreateDto {
  LineId?: number;
  HeaderId: number;
}

export interface UpdateGrImportLDto extends BaseImportLineUpdateDto {
  LineId?: number;
  HeaderId?: number;
}

